modules/core/src/test/scala/org/tessellation/infrastructure/rewards/RewardsSuite.scala

```diff
@@ -0,0 +1,135 @@
+package org.tessellation.infrastructure.rewards
+
+import cats.data.NonEmptySet
+import cats.effect.std.Random
+import cats.effect.{IO, Resource}
+import cats.implicits.catsSyntaxApplicativeId
+import cats.syntax.contravariantSemigroupal._
+
+import scala.collection.immutable.SortedSet
+import scala.concurrent.duration.DurationInt
+
+import org.tessellation.config.types._
+import org.tessellation.dag.dagSharedKryoRegistrar
+import org.tessellation.dag.snapshot.epoch.EpochProgress
+import org.tessellation.ext.kryo._
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.schema.ID.Id
+import org.tessellation.sdk.sdkKryoRegistrar
+import org.tessellation.security.SecurityProvider
+import org.tessellation.security.hash.Hash
+import org.tessellation.security.hex.Hex
+
+import eu.timepit.refined.auto._
+import weaver.MutableIOSuite
+import weaver.scalacheck.Checkers
+
+object RewardsSuite extends MutableIOSuite with Checkers {
```

Comment:
I haven't review this test yet, so there might be additional comments regarding it.